### ISSUE TITLE
Connections not working with from_definition

### ISSUE THREAD

**Issue by jmondo** [rmosolgo/graphql-ruby#2731]
As part of testing a feature of our app, we create a temporary graphql schema for our test suite.
We are using `GraphQL::Schema.from_definition` to build this schema.

Pretty simple schema:
```
type Query {
    posts(authorIds: [ID!], postTypeIds: [ID!]): [Post]
    postConnection(
        after: String
        before: String
        first: Int
        last: Int
    ): PostConnection!
  }

  type Mutation {
    upvotePost (
      postId: ID!
    ): Post
  }

  type PostConnection {
    edges: [PostEdge]
    pageInfo: PageInfo!
  }

  type PostEdge {
    cursor: String!
    node: Post
  }

  type PageInfo {
    endCursor: String
    hasNextPage: Boolean!
    hasPreviousPage: Boolean!
    startCursor: String
  }
```

I found that by naming PostConnection end with Connection, there's code to turn on Connection support.  https://github.com/rmosolgo/graphql-ruby/blob/6c4dc67be6a1049caf5279b722e2ac5c6149e0c8/lib/graphql/schema/build_from_definition.rb#L318 

But when I tried to use the schema, fetching the node from each edge was failing. 
After lots of digging I found that's because PostEdge wasn't being set up as an edge.
It seems like the code I linked above needs an additional check for classes that end with "Edge". In those cases, set edge_class to Relay::Edge.
Without that, it seems the Connection support is half enabled.

That being said, for the purposes of our feature, we'd actually like to turn off connection support entirely and put something like this (below) in our DefaultResolver. I'm not sure how common that need would be for other users, but it would be nice to have that option to pass to `from_definition`

```
when 'postConnection'
{
  "edges" => [
    {
      "cursor": 'fakecursor',
      "node": {"id"=>1, "postTypeId"=>1, "authorId"=>1, "title"=>"Introduction to GraphQL", "votes"=>2}
    }
  ],
  "pageInfo" => {
    ...
  }
}
```

For now we've just monkeypatched out this line https://github.com/rmosolgo/graphql-ruby/blob/6c4dc67be6a1049caf5279b722e2ac5c6149e0c8/lib/graphql/schema/build_from_definition.rb#L318 and it's working, but I thought it might be helpful to report this.

**Comment by rmosolgo** [comment]
Yes, thanks for reporting! I'll look into properly enabling Edge detection. Glad you found a work-around for now.
